org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/XMLModelDiagnosticsTest.java

```diff
@@ -80,6 +83,36 @@ public void xmlModelWithXSDAndNamespace() throws Exception {
 		testDiagnosticsFor(xml, d);
 	}
 
+	@Test
+	public void cvc_elt_1_a_basic() throws Exception {
+		String xml = "<test xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\r\n" + //
+				"    xsi:noNamespaceSchemaLocation=\"src/test/resources/xsd/unique.xsd\">\r\n" + //
+				"    <authors status = \"new\">\r\n" + //
+				"    <author>smith</author>\r\n" + //
+				"    </authors>\r\n" + //
+				"</test>";
+		Diagnostic d = d(0, 1, 0, 5, XMLSchemaErrorCode.cvc_elt_1_a);
+		testDiagnosticsFor(xml, d);
+		testCodeActionsFor(xml, d, ca(d, te(0, 1, 0, 5, "root"),
+				te(5, 2, 5, 6, "root")));
+	}
+
+	@Test
+	public void cvc_elt_1_a_multiple_roots_defined() throws Exception {
+		String xml = "<test xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\r\n" + //
+				"    xsi:noNamespaceSchemaLocation=\"src/test/resources/xsd/unique_multiroot.xsd\">\r\n" + //
+				"    <authors status = \"new\">\r\n" + //
+				"    <author>smith</author>\r\n" + //
+				"    </authors>\r\n" + //
+				"</test>";
+		Diagnostic d = d(0, 1, 0, 5, XMLSchemaErrorCode.cvc_elt_1_a);
+		testDiagnosticsFor(xml, d);
+		testCodeActionsFor(xml, d, ca(d, te(0, 1, 0, 5, "root"),
+				te(5, 2, 5, 6, "root")),
+				ca(d, te(0, 1, 0, 5, "root2"),
+						te(5, 2, 5, 6, "root2")));
+	}
+

```

Comment:
Could you write a test with xml which have no end tag. By reading your code it seems you take care about that but we need test this usecase.

Reply:
My code currently doesn't add a new end tag if there is no end tag. Should I extend the code action to also insert the correct end tag if it doesn't exist?

Reply:
No your behavior is perfect, but a test wth this usecase will prevent from NPE or other bug if we change your code. We will sure that we cover all usecases.

It is difficult to create an end tag, because the question is where you create the end tag?

Reply:
Thanks @JessicaJHee for the test.